tool/site/client-react/src/pages/returnPage.tsx

```diff
@@ -0,0 +1,217 @@
+    return (
+      <>
+        <Label isPageHeading>Return prescription</Label>
+        <ReturnForm prescriptionId={prescriptionId} onSubmit={setReturnFormValues} />
+      </>
+    )
+  }
+  const sendReturnTask = () => sendReturn(baseUrl, returnFormValues)
+  return (
+    <LongRunningTask<ApiResult> task={sendReturnTask} loadingMessage="Sending return.">
+      {returnResult => (
+        <>
+          <Label isPageHeading>Return Result {returnResult.success ? <TickIcon /> : <CrossIcon />}</Label>
+          <MessageExpanders
+            fhirRequest={returnResult.request}
+            hl7V3Request={returnResult.request_xml}
+            fhirResponse={returnResult.response}
+            hl7V3Response={returnResult.response_xml}
+          />
+          <ButtonList>
+            <ReloadButton />
+          </ButtonList>
+        </>
+      )}
+    </LongRunningTask>
+  )
+}
+
+interface ReturnFormProps {
+  prescriptionId?: string
+  onSubmit: (values: ReturnFormValues) => void
+}
+
+const ReturnForm: React.FC<ReturnFormProps> = ({
+  prescriptionId,
+  onSubmit
+}) => {
+  const initialValues: ReturnFormValues = {prescriptionId, pharmacy: "VNFKT", reason: "0002"}
+  return (
+    <>
+      <SummaryList>
+        <SummaryList.Row>
+          <SummaryList.Key>ID</SummaryList.Key>
+          <SummaryList.Value>{prescriptionId}</SummaryList.Value>
+        </SummaryList.Row>
+      </SummaryList>
+      <Formik<ReturnFormValues> initialValues={initialValues} onSubmit={values => onSubmit(values)}>
+        {formik => <Form onSubmit={formik.handleSubmit} onReset={formik.handleReset}>
+          <Fieldset>
+            <PharmacyRadios
+              label="Pharmacy returning prescription"
+              defaultValue={initialValues.pharmacy}
+              value={formik.values.pharmacy}
+              error={formik.errors.pharmacy} />
+            <RadioField
+              name="reason"
+              label="Choose a reason for returning"
+              defaultValue={initialValues.reason}
+              fieldRadios={returnReasons} />
+          </Fieldset>
+          <ButtonList>
+            <Button type="submit">Return</Button>
+            <BackButton />
+          </ButtonList>
+        </Form>}
+      </Formik>
+    </>
+  )
+}
+
+async function sendReturn(
+  baseUrl: string,
+  returnFormValues: ReturnFormValues
+): Promise<ApiResult> {
+  const returnParameters = createReturn(returnFormValues)
+  const returnResponse = await axiosInstance.post<ApiResult>(`${baseUrl}dispense/return`, returnParameters)
+  return getResponseDataIfValid(returnResponse, isApiResult)
+}
+
+function createReturn(returnFormValues: ReturnFormValues): fhir.Task {
+  const identifier = uuid.v4()
+  const bundleIdentifier = identifier
+
+  return {
+    resourceType: "Task",
+    id: identifier,
+    identifier: [
+      {
+        system: "https://tools.ietf.org/html/rfc4122",
+        value: bundleIdentifier
+      }
+    ],
+    status: "rejected",
```

Comment:
[apparently](https://simplifier.net/guide/NHSDigital-Medicines/Home/FHIRAssets/AllAssets/Profiles/NHSDigital-Task.guide.md#status), status can be one of "rejected", "cancelled" or "failed" for a return, is it worth having that editable in the UI?

Reply:
I think generally we should carry on following the approach to add minimum required editable fields until we have requests otherwise. Can always add this in if we get a requirement to in future, but the added value here is being able to return and re-release prescriptions